lib/commands.coffee

```diff
@@ -125,18 +129,51 @@ module.exports =
 	burn: (image, operation, options) ->
 		# Default image to the given path
 		image = operation.image ? image
+		emitter = new EventEmitter()
 
 		Promise.try ->
 			if not options?.drive?
 				throw new Error('Missing drive option')
 
+			file = new sdk.sourceDestination.File({
+				path: image
+			})
 			Promise.props
-				drive: normalizeDrive(options.drive)
-				imageSize: fs.statAsync(image).get('size')
-				imageStream: fs.createReadStream(image)
-		.then ({ drive, imageStream, imageSize }) ->
-			imageWrite.write drive,
-				stream: imageStream
-				size: imageSize
-			,
-				check: true
+				drive: getDrive(options.drive)
+				source: file.getInnerSource()
+		.then ({ drive, source }) ->
+			start = Date.now()
+			progressState = {
+				transferred: 0
+			}
+			sdk.multiWrite.pipeSourceToDestinations({
+				source,
+				destinations: [drive],
+				onFail: (_, error) -> emitter.emit('error', error)
+				onProgress: (progress) ->
+					type = null
+					if progress.type == 'flashing'
+						type = 'write'
+					if progress.type == 'verifying'
+						type = 'check'
+					if not type?
+						return
+
+					progress.type = type
+					progressState = {
```

Comment:
Tries to emulate what etcher-image-write was emitting
See: https://github.com/balena-io-modules/etcher-image-write/blob/03862b6f7685f7540b8f8d0c89f8d5c02175900c/lib/index.js#L147